### PR TITLE
Feat/1174 token inputs

### PR DESCRIPTION
**Pull request by akdetrick**
relates to #1174 

- Adds `FieldToken`
- Adds `TokenInput`

### CODE REVIEW

**Comment by akdetrick** [comment]
I guess I should probably put a max width on tokens

Screenshot 2024-04-30 at 1 49 34 PM

**Comment by akdetrick** [comment]
If the max width of `300px` for `FieldToken` becomes a problem, we can always change `startContent` in `Input` to have a max width with overflow hidden.

**Comment by agreatscott** [comment]
I might be doing something wrong, but it looks like when I try to add new values to the input in the story, I can't get them to convert to tokens. But when I hardcode the value in they are of course shown as tokens, is it just that this is expected to be a controlled component? We almost always use controlled components for these inputs so I think this might be fine. 

image

**Comment by agreatscott** [comment]
> `TokenInput` is a **fully controlled** component
 
 Ah, I just can't read! Disregard my above comment

File: src/index.scss [hunk 1]
```diff
@@ -1,5 +1,10 @@
 @use "sass:map";
 
+// 💁 Imports are ordered by least specific to most specific.
+//    For example, helper classes should be further down in the cascade
+//    than base styles, so that selectors of the same specificity will
+//    resolve based on source order (last wins).
+
```
Comment (agreatscott) [inline src/index.scss:7]: 🧠

File: src/TokenInput/index.jsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+import React, { useRef } from "react";
+import PropTypes from "prop-types";
+import FieldToken from "../FieldToken";
+import TextInput from "../TextInput";
+
+const noop = () => {};
+const MIN_TOKEN_LENGTH = 1; // assume anything less than 2 is an accident
+const INPUT_RESET_EVENT = { target: { value: "" } };
+
+// take value of input with any leading or trailing delimiters removed
```
Comment (akdetrick) [inline src/TokenInput/index.jsx:15]:
```suggestion
// take value of input with leading and trailing delimiters removed
```

**Review by agreatscott — APPROVED** [review]
discussed renaming some vars and accessibility tweaks offline - lgtm!

**Pull request merged**